content/docs/for-buildpack-authors/how-to/write-buildpacks/clear-env.md

```diff
@@ -3,8 +3,23 @@ title="Clear the buildpack environment"
 weight=99
 +++
 
+During the `detect` and `build` phases, the lifecycle MUST provide as environment variables any user-provided files in `<platform>/env/` with environment variable names and contents matching the file names and contents.
```

Comment:
The `<platform>/env/` directory is just the mechanism by which the platform can provide build-time environment variables for buildpacks to the lifecycle, so the lifecycle can set those up. Buildpack authors don't really have to care that it's called `<platform>/env/`, just that there is a way for end users to specify build-time environment variables. "Cear env" is a way for a buildpack to declare "I am not configurable by the end user". 

Note that buildpacks CANNOT declare "I am not configurable by the platform". That is a separate thing.

Reply:
I think somewhere in here we want to tell the buildpack author how they can elect to clear the environment. It is by setting `clear-env` to true in buildpack.toml: https://github.com/buildpacks/spec/blob/main/buildpack.md#buildpacktoml-toml

Reply:
We should also mention, for buildpacks that DO allow themselves to be configured by the end user, there is a special convention for that:

https://github.com/buildpacks/spec/blob/main/buildpack.md#provided-by-the-platform

i.e., `BP_*`, and `BPL_*`. Just as an example, here is a Paketo buildpack that exposes 3 configuration variables that all have the prefix: https://github.com/paketo-buildpacks/ca-certificates?tab=readme-ov-file#configuration